ansible_collections/arista/avd/roles/eos_designs/templates/designs/l3ls-evpn/underlay/ebgp/spine-router-bgp.j2

```diff
@@ -19,68 +15,71 @@ router_bgp:
-{%             set leaf_spines = l3leaf.node_groups[l3leaf_node_group].spines %}
-{%         else %}
-{%             set leaf_spines = l3leaf.defaults.spines %}
-{%         endif %}
-{%         for node in l3leaf.node_groups[l3leaf_node_group].nodes | arista.avd.natural_sort %}
-{# Underlay network peering #}
-{%             for leaf_spine in leaf_spines %}
-{%                 if leaf_spine == inventory_hostname %}
+{% endif %}
+{% for l3leaf_node_group in l3leaf.node_groups | arista.avd.natural_sort %}
+{%     if l3leaf.node_groups[l3leaf_node_group].spines is defined %}
+{%         set leaf_spines = l3leaf.node_groups[l3leaf_node_group].spines %}
+{%     else %}
+{%         set leaf_spines = l3leaf.defaults.spines %}
+{%     endif %}
+{%     for node in l3leaf.node_groups[l3leaf_node_group].nodes | arista.avd.natural_sort %}
+{#erlay network peering #}
+{%         for leaf_spine in leaf_spines %}
+{%             if leaf_spine == inventory_hostname %}
+{%                 if underlay_rfc5549 is arista.avd.defined(true) %}
+    {{ l3leaf.node_groups[l3leaf_node_group].nodes[node].spine_interfaces[loop.index0] }}:
+{%                 else %}
     {{ underlay_p2p_network_summary | ipaddr('network') | ipmath(((l3leaf.node_groups[l3leaf_node_group].nodes[node].id -1) * 2 * switch.max_spines * max_l3leaf_to_spine_links) + loop.index0 * 2 + 1) }}:
+{%                 endif %}
       peer_group: {{ bgp_peer_groups.IPv4_UNDERLAY_PEERS.name | arista.avd.default("IPv4-UNDERLAY-PEERS") }}
       remote_as: {{ l3leaf.node_groups[l3leaf_node_group].nodes[node].bgp_as | arista.avd.default(
                     l3leaf.node_groups[l3leaf_node_group].bgp_as,
                     l3leaf.defaults.bgp_as) }}
       description: {{ node }}_{{ l3leaf.node_groups[l3leaf_node_group].nodes[node].spine_interfaces[loop.index0] }}
-{%                 endif %}
-{%             endfor %}
+{%             endif %}
 {%         endfor %}
 {%     endfor %}
+{% endfor %}
 
 {# Underlay network peering to super spines #}
-{%     set spine_uplink_list = spine.nodes[inventory_hostname]['uplinks_to_super_spine_interfaces'] | arista.avd.default(
-                                   spine.uplinks_to_super_spine_interfaces,
-                                   []) %}
-{%     for spine_uplink in spine_uplink_list %}
-{%         set s_spine_id = loop.index0 // max_spine_to_super_spine_links + 1 %}
-{%         set ip_pool = super_spine_underlay_p2p_network_summary %}
-{#     max number of hosts in subnet will be divided by max super-spine number to avoid address changes when provisioning new spines #}
-{%         set ip_pool_max_hosts_in_subnet = ip_pool | ipaddr('size') %}
-{%         set offset = loop.index0 % max_spine_to_super_spine_links %}
+{% set spine_uplink_list = spine.nodes[inventory_hostname]['uplinks_to_super_spine_interfaces'] | arista.avd.default(
+                               spine.uplinks_to_super_spine_interfaces,
```

Comment:
Seems this line is over-indented and should have few spaces less to align `spine` with line 51

Reply:
We will keep it as-is for now and discuss how to refactor afterwards.